Source/WebCore/html/HTMLMediaElement.cpp

```diff
@@ -5702,18 +5702,32 @@ void HTMLMediaElement::mediaPlayerSizeChanged()
     endProcessingMediaPlayerCallback();
 }
 
-bool HTMLMediaElement::mediaPlayerRenderingCanBeAccelerated()
+void HTMLMediaElement::invalidateStyleAndLayerComposition()
 {
+    HTMLElement::invalidateStyleAndLayerComposition();
+    updateAcceleratedRenderingState();
```

Comment:
In general it's not a good design to compute state in an "invalidate" function. This kind of function should just mark some state as dirty, and that state should be computed (possibly at some cost) later at layout/compositing time, which is what the current design does. For example, how does this code take into account whether the poster is showing?( (A video showing the poster should not get a compositing layer.)